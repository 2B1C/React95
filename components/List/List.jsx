import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Field = styled.ul``;

class List extends React.Component {
  render() {
    return <Field />;
  }
}

List.propTypes = {};

List.defaultProps = {};

export default List;
