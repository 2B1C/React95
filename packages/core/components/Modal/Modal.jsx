import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';
import Draggable from 'react-draggable';

import Btn from '../shared-style/Btn';
import Button from '../Button';
import Icon from '../Icon';

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;

  position: fixed;

  padding: 2 2 8;
  ${({ width, height }) => `width: ${width}px; height: ${height}px;`}

  top: 50px;

  background-color: bg;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
    inset 0 0 0 1px ${th('colors.grays.3')}, 1px 1px 0 1px ${th('colors.black')};
`;

const TitleBar = styled.div`
  height: 18px;
  margin-bottom: 2;

  background-color: primary;
  color: ${th('colors.white')};
  padding: 2 2 0;

  display: flex;
`;

const Title = styled.div`
  flex-grow: 1;
  font-weight: bold;
`;

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Btn)`
  margin-right: 2;
  padding: 0;

  width: 17px;
  height: 14px;
  min-width: 0;

  font-size: 10;

  &:last-child {
    margin-right: 0;
  }

  &:active {
    padding: 1 0 0 1;

    outline: none;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
      inset -1px -1px 0px 1px ${th('colors.grays.3')};
  }
`;

Option.displayName = 'Option';

const Content = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  padding: 6;
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.buttonsAlignment};

  padding: 0 6 6 6;

  & ${Btn} {
    margin-right: 6;
    min-width: 70px;

    &:last-child {
      margin-right: 0;
    }
  }
`;

const MenuWrapper = styled.ul`
  display: flex;
  flex-direction: row;

  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3;

  border-bottom-style: solid;
  border-width: 1;
  border-bottom-color: grays.3;

  box-shadow: 0 1px 0 0 ${th('colors.grays.0')};
`;

const MenuItem = styled.li`
  position: relative;
  padding-left: 6;
  padding-right: 6;

  user-select: none;

  ul {
    position: absolute;
    left: 0;
    color: ${th('colors.black')};
  }

  ${({ active }) =>
    active &&
    css`
      background-color: primary;
      color: ${th('colors.white')};
    `};
`;

MenuItem.displayName = 'MenuItem';

const Modal = ({
  closeModal,
  title,
  children,
  buttons,
  icon,
  menu,
  buttonsAlignment,
  defaultPosition,
  width,
  height,
  ...rest
}) => {
  const [menuOpened, setMenuOpened] = useState('');

  const iconStyle = {
    width: 15,
    height: 13,
    style: {
      marginRight: '4px',
    },
  };

  return (
    <>
      <Draggable handle=".draggable" defaultPosition={defaultPosition}>
        <ModalWrapper width={width} height={height} {...rest}>
          <TitleBar className="draggable">
            {icon && <Icon name={icon} {...iconStyle} />}
            <Title>{title}</Title>
            <OptionsBox>
              <Option>?</Option>
              <Option onClick={closeModal}>x</Option>
            </OptionsBox>
          </TitleBar>

          {menu.length > 0 && (
            <MenuWrapper>
              {menu.map(({ name, list }) => {
                const active = menuOpened === name;
                return (
                  <MenuItem
                    key={name}
                    onMouseDown={() => setMenuOpened(name)}
                    active={active}
                  >
                    {name}
                    {active && list}
                  </MenuItem>
                );
              })}
            </MenuWrapper>
          )}

          <Content onClick={() => setMenuOpened('')}>{children}</Content>
          {buttons.length > 0 && (
            <ButtonWrapper buttonsAlignment={buttonsAlignment}>
              {buttons.map((button, index) => (
                <Button
                  // eslint-disable-next-line react/no-array-index-key
                  key={index}
                  onClick={button.onClick}
                  value={button.value}
                />
              ))}
            </ButtonWrapper>
          )}
        </ModalWrapper>
      </Draggable>
    </>
  );
};

Modal.displayName = 'Modal';

Modal.propTypes = {
  icon: PropTypes.string,
  closeModal: PropTypes.func,
  title: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      onClick: PropTypes.func,
    }),
  ),
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      list: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
      ]).isRequired,
    }),
  ),
  buttonsAlignment: PropTypes.oneOf(['center', 'flex-start', 'flex-end']),
  defaultPosition: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
  }),
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Modal.defaultProps = {
  icon: '',
  title: 'Modal',
  buttonsAlignment: 'flex-end',
  children: null,
  defaultPosition: { x: 0, y: 0 },
  buttons: [],
  menu: [],
  width: 300,
  height: 200,
  closeModal: () => {},
};

export default Modal;
