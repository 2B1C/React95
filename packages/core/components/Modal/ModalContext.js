import { createContext } from 'react';

const ModalContext = createContext({ windows: [] });

export default ModalContext;
