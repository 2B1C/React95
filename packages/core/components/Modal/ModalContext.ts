/* eslint-disable no-unused-vars */
import { createContext } from 'react';

import { IconProps } from '../Icon/Icon';

export type Windows = {
  icon?: IconProps['name'];
  title: string;
};

export interface IModalContextProps {
  windows: Array<Windows>;
  addWindows(window: Windows): void;
  removeWindows(title: string): void;
  setActiveWindow(title: string): void;
  activeWindow?: string;
}

const ModalContext = createContext<IModalContextProps>({
  windows: [],
  addWindows: () => {},
  removeWindows: () => {},
  setActiveWindow: () => {},
  activeWindow: '',
});

export default ModalContext;
