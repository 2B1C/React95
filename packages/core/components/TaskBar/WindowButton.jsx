import React from 'react';
import PropTypes from 'prop-types';

import Frame from '../Frame';
import Icon from '../Icon';

const WindowButton = ({ children, small, icon, active, ...props }) => (
  <Frame
    style={{
      display: 'flex',
      justifyContent: 'flex-start',
      alignItems: 'center',
      padding: '2px 3px',
      maxWidth: 150,
      width: small ? undefined : '100%',
    }}
    {...(active
      ? {
          boxShadow: 'in',
          bg: 'grays.0',
        }
      : {
          boxShadow: 'out',
        })}
    {...props}
    mr={2}
  >
    <Icon name={icon} style={{ marginRight: 4 }} width={20} height={20} />
    {children}
  </Frame>
);

WindowButton.propTypes = {
  children: PropTypes.node,
  small: PropTypes.bool,
  icon: PropTypes.string,
  active: PropTypes.bool,
};

WindowButton.defaultProps = {
  children: '',
  small: false,
  icon: '',
  active: false,
};

export default WindowButton;
