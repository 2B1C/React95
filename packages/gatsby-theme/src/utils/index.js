import isEmpty from './isEmpty';
import navify from './navify';

export { isEmpty, navify };
