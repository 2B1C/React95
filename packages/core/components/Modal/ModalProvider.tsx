import { nanoid } from 'nanoid';
import * as React from 'react';

import ModalContext, { Windows } from './ModalContext';

// eslint-disable-next-line @typescript-eslint/ban-types
export type ModalProviderProps = {};

type WindowStack = Record<string, Windows>;

type WindowAction =
  | {
      type: 'ADD_WINDOW';
      id: string;
      window: Windows;
    }
  | {
      type: 'UPDATE_WINDOW';
      id: string;
      window: Windows;
    }
  | {
      type: 'REMOVE_WINDOW';
      id: string;
    }
  | {
      type: 'REMOVE_BY_TITLE';
      title: string;
    };

const windowStackReducer: React.Reducer<WindowStack, WindowAction> = (
  state,
  action,
) => {
  const newWindows = { ...state };
  switch (action.type) {
    case 'ADD_WINDOW': {
      newWindows[action.id] = action.window;
      return newWindows;
    }
    case 'REMOVE_WINDOW': {
      delete newWindows[action.id];
      return newWindows;
    }
    case 'UPDATE_WINDOW': {
      newWindows[action.id] = action.window;
      return newWindows;
    }
    case 'REMOVE_BY_TITLE': {
      Object.entries(newWindows).forEach(([id, window]) => {
        if (window.title === action.title) {
          delete newWindows[id];
        }
      });
      return newWindows;
    }
    default:
      return state;
  }
};

const ModalProvider: React.FunctionComponent = ({ children }) => {
  const [windows, dispatch] = React.useReducer(windowStackReducer, {});
  const [activeWindow, setActiveWindow] = React.useState<string>();

  const addWindows = (window: Windows) => {
    const id = nanoid();
    dispatch({ type: 'ADD_WINDOW', id, window });
    return id;
  };
  // TODO: Remove this method in the next major release - updating by IDs is more sensible
  const removeWindows = (title: string) => {
    dispatch({ type: 'REMOVE_BY_TITLE', title });
  };
  const removeWindowById = (id: string) => {
    dispatch({ type: 'REMOVE_WINDOW', id });
  };
  const updateWindow = (id: string, window: Windows) => {
    dispatch({ type: 'UPDATE_WINDOW', id, window });
  };

  return (
    <ModalContext.Provider
      value={{
        windows,
        addWindows,
        removeWindows,
        removeWindowById,
        updateWindow,
        setActiveWindow,
        activeWindow,
      }}
    >
      {children}
    </ModalContext.Provider>
  );
};

export default ModalProvider;
