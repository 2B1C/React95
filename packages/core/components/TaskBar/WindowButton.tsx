import React from 'react';
import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';

import Frame, { FrameProps } from '../Frame/Frame';
import Icon from '../Icon';
import { Windows } from '../Modal/ModalContext';

export type WindowButtonProps = {
  small?: boolean;
  active?: boolean;
} & Pick<Windows, 'icon'> &
  React.HTMLAttributes<HTMLButtonElement> &
  FrameProps;

const Button = styled(Frame)<Omit<WindowButtonProps, 'icon'>>`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding: 2 3;
  margin-right: 2;
  max-width: 150px;
  border: none;
  outline: none;

  ${({ small }) =>
    !small
      ? `
    width: 100%;
  `
      : ''}

  ${({ active, small }) =>
    active && small
      ? css`
          outline: ${th('space.1')}px dotted ${th('colors.black')};
          outline-offset: -${th('space.4')}px;
          padding-top: 4;
          padding-right: 2;
          padding-bottom: 0;
          padding-left: 4;
        `
      : ''}
`;

const WindowButton: React.FC<WindowButtonProps> = ({
  children = '',
  small,
  icon = undefined,
  active,
  ...props
}) => (
  <Button
    active={active}
    small={small}
    {...(active
      ? {
          boxShadow: 'in',
          bg: small ? undefined : 'grays.0',
        }
      : {
          boxShadow: 'out',
        })}
    {...props}
    as="button"
  >
    <Icon name={icon} style={{ marginRight: 4 }} width={20} height={20} />
    {children}
  </Button>
);

WindowButton.defaultProps = {
  small: false,
  icon: undefined,
  active: false,
};

export default WindowButton;
