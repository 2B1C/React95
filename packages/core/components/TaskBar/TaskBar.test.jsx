import React from 'react';
import { render, fireEvent, act } from '../shared/test/utils';

import { Modal } from '../Modal';
import List from '../List';

import Clock from './Clock';
import WindowButton from './WindowButton';
import TaskBar from './TaskBar';

describe('<TaskBar />', () => {
  beforeAll(() => jest.useFakeTimers());
  afterAll(() => jest.useRealTimers());

  describe('Snapshots', () => {
    describe('<Clock />', () => {
      it('should match snapshot', () => {
        const { container } = render(<Clock />);
        expect(container).toMatchSnapshot();
      });
    });

    describe('<WindowButton />', () => {
      it('should match snapshot', () => {
        const { container } = render(<WindowButton />);
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with small prop', () => {
        const { container } = render(<WindowButton small />);
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with active prop', () => {
        const { container } = render(
          <>
            <WindowButton small />
            <WindowButton small active />
          </>,
        );
        expect(container).toMatchSnapshot();
      });

      it('should match snapshot with icon prop', () => {
        const { container } = render(<WindowButton icon="bat" />);
        expect(container).toMatchSnapshot();
      });
    });

    it('should match snapshot', () => {
      act(() => jest.advanceTimersByTime(1000));

      const { container } = render(<TaskBar />);

      expect(container).toMatchSnapshot();
    });
  });
});
