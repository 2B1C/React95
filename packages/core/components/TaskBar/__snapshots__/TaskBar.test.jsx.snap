// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskBar /> Snapshots <Clock /> should match snapshot 1`] = `
.c0 {
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 2px;
  padding-bottom: 2px;
  background-color: transparent;
  box-shadow: inset 0px 0px 0px 0px, inset 1px 1px 0px 0px #868a8e, 0.5px 0.5px 0px 0.5px #ffffff;
}

.c3 {
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  background: radial-gradient(#ff0 20%,transparent 20%) 0 0, radial-gradient(#ff0 20%,transparent 20%) 4px 4px, radial-gradient(rgba(255,255,255,0.1) 20%,transparent 25%) 0 1px, radial-gradient(rgba(255,255,255,0.1) 20%,transparent 25%) 3px 4px;
  background-size: 7px 7px;
  background-color: #ffffff;
  border: 1px solid;
  padding: 2px 2px 2px 4px;
  box-shadow: none;
  position: absolute;
  top: -20px;
  text-align: center;
  z-index: 100;
  display: none;
}

.c1 {
  display: inline-block;
  position: relative;
  cursor: default;
  white-space: nowrap;
}

.c2 div:first-child {
  right: 0;
}

<div>
  <div
    class="c0"
    style="display: flex; justify-content: center; align-items: center;"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        30 March 2020
      </div>
      
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> Snapshots <WindowButton /> should match snapshot 1`] = `
.c1 {
  display: block;
  width: 20px;
  height: 20px;
  background-image: url('none');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  margin-right: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

<div>
  <div
    class="c0"
    style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px; width: 100%;"
  >
    <i
      class="c1"
      height="20"
      name=""
      style="margin-right: 4px;"
      width="20"
    />
    
  </div>
</div>
`;

exports[`<TaskBar /> Snapshots <WindowButton /> should match snapshot with active prop 1`] = `
.c1 {
  display: block;
  width: 20px;
  height: 20px;
  background-image: url('none');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  margin-right: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

.c2 {
  margin-right: 2px;
  background-color: #e6e6e6;
  box-shadow: inset 0px 0px 0px 0px, inset 1px 1px 0px 0px #868a8e, 0.5px 0.5px 0px 0.5px #ffffff;
}

<div>
  <div
    class="c0"
    style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px;"
  >
    <i
      class="c1"
      height="20"
      name=""
      style="margin-right: 4px;"
      width="20"
    />
    
  </div>
  <div
    class="c2"
    style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px;"
  >
    <i
      class="c1"
      height="20"
      name=""
      style="margin-right: 4px;"
      width="20"
    />
    
  </div>
</div>
`;

exports[`<TaskBar /> Snapshots <WindowButton /> should match snapshot with icon prop 1`] = `
.c1 {
  display: block;
  width: 20px;
  height: 20px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  margin-right: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

<div>
  <div
    class="c0"
    style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px; width: 100%;"
  >
    <i
      class="c1"
      height="20"
      name="bat"
      style="margin-right: 4px;"
      width="20"
    />
    
  </div>
</div>
`;

exports[`<TaskBar /> Snapshots <WindowButton /> should match snapshot with small prop 1`] = `
.c1 {
  display: block;
  width: 20px;
  height: 20px;
  background-image: url('none');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  margin-right: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

<div>
  <div
    class="c0"
    style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px;"
  >
    <i
      class="c1"
      height="20"
      name=""
      style="margin-right: 4px;"
      width="20"
    />
    
  </div>
</div>
`;

exports[`<TaskBar /> Snapshots should match snapshot 1`] = `
.c2 {
  display: block;
  width: 20px;
  height: 20px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  padding: 2px;
  width: 100%;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

.c1 {
  margin-right: 2px;
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
}

.c3 {
  padding-left: 0;
  margin-left: 2px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #c3c7cb;
  box-shadow: none;
}

.c4 {
  padding-left: 6px;
  padding-right: 6px;
  padding-top: 2px;
  padding-bottom: 2px;
  background-color: transparent;
  box-shadow: inset 0px 0px 0px 0px, inset 1px 1px 0px 0px #868a8e, 0.5px 0.5px 0px 0.5px #ffffff;
}

.c7 {
  background-color: #c3c7cb;
  box-shadow: inset 0.5px 0.5px 0px 0.5px #ffffff, inset 0 0 0 1px #868a8e, 1px 0px 0 0px #000000, 0px 1px 0 0px #000000, 1px 1px 0 0px #000000;
  background: radial-gradient(#ff0 20%,transparent 20%) 0 0, radial-gradient(#ff0 20%,transparent 20%) 4px 4px, radial-gradient(rgba(255,255,255,0.1) 20%,transparent 25%) 0 1px, radial-gradient(rgba(255,255,255,0.1) 20%,transparent 25%) 3px 4px;
  background-size: 7px 7px;
  background-color: #ffffff;
  border: 1px solid;
  padding: 2px 2px 2px 4px;
  box-shadow: none;
  position: absolute;
  top: -20px;
  text-align: center;
  z-index: 100;
  display: none;
}

.c5 {
  display: inline-block;
  position: relative;
  cursor: default;
  white-space: nowrap;
}

.c6 div:first-child {
  right: 0;
}

<div>
  <div
    class="c0"
    style="position: fixed; bottom: 0px; left: 0px; right: 0px; display: flex; justify-content: space-between; height: 28px;"
    width="100%"
  >
    <div
      class="c1"
      style="display: flex; justify-content: flex-start; align-items: center; padding: 2px 3px; max-width: 150px;"
    >
      <i
        class="c2"
        height="20"
        name="logo"
        style="margin-right: 4px;"
        width="20"
      />
      Start
    </div>
    <div
      class="c3"
      display="flex"
      width="100%"
    />
    <div
      class="c4"
      style="display: flex; justify-content: center; align-items: center;"
    >
      <div
        class="c5 c6"
      >
        <div
          class="c7"
        >
          30 March 2020
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
